### PR TITLE
fix : Max inactive interval ignored

### PR DESCRIPTION
**Pull request by vignjatov**
fix for #209 Max inactive interval ignored

### CODE REVIEW

**Comment by graemerocher** [comment]
Could you do the commits with the same account as the PR is created with?

**Comment by vignjatov** [comment]
Added new commit with the same user that created PR.

File: redis-lettuce/src/main/java/io/micronaut/configuration/lettuce/session/RedisSessionStore.java [hunk 1]
```diff
@@ -281,9 +281,16 @@ public RedisSession newSession() {
 
     @Override
     public CompletableFuture<Optional<RedisSession>> findSession(String id) {
-        return findSessionInternal(id, false);
+        return findSessionInternal(id, false).thenCompose(session -> {
+            if (session.isPresent()) {
+                RedisSession redisSession = session.get();
+                redisSession.setLastAccessedTime(Instant.now());
+            }
```
Comment (graemerocher) [inline redis-lettuce/src/main/java/io/micronaut/configuration/lettuce/session/RedisSessionStore.java:288]: can be written as:

```java
session.ifPresent(redisSession -> 
    redisSession.setLastAccessedTime(Instant.now()
)
```

File: redis-lettuce/src/main/java/io/micronaut/configuration/lettuce/session/RedisSessionStore.java [hunk 2]
```diff
@@ -281,9 +281,16 @@ public RedisSession newSession() {
 
     @Override
     public CompletableFuture<Optional<RedisSession>> findSession(String id) {
-        return findSessionInternal(id, false);
+        return findSessionInternal(id, false).thenCompose(session -> {
```
Comment (graemerocher) [inline redis-lettuce/src/main/java/io/micronaut/configuration/lettuce/session/RedisSessionStore.java:284]: Easier to use `thenApply` and just return the `session` inside of wrapping again in `CompletableFuture.completedFuture`

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: redis-lettuce/src/main/java/io/micronaut/configuration/lettuce/session/RedisSessionStore.java (hunk 1 above)
Comment (vignjatov) [reply redis-lettuce/src/main/java/io/micronaut/configuration/lettuce/session/RedisSessionStore.java]: agree, updated the code.

File: redis-lettuce/src/main/java/io/micronaut/configuration/lettuce/session/RedisSessionStore.java (hunk 2 above)
Comment (vignjatov) [reply redis-lettuce/src/main/java/io/micronaut/configuration/lettuce/session/RedisSessionStore.java]: updated with proposed changes.

**Comment by graemerocher** [comment]
Thanks for the contribution!

**Pull request merged**